### PR TITLE
[ros2node] Add option to info verb to display hidden names

### PR DESCRIPTION
**Pull request by jacobperron**
Changes behavior so that hidden names are not shown by default.

Before this change, if we ran turtlesim and retrieved the node info we'd see hidden topics and services related to the action server, e.g.

    $ ros2 run turtlesim turtlesim_node

```sh
$ ros2 node info /turtlesim
/turtlesim
  Subscribers:
    /parameter_events: rcl_interfaces/msg/ParameterEvent
    /turtle1/cmd_vel: geometry_msgs/msg/Twist
  Publishers:
    /parameter_events: rcl_interfaces/msg/ParameterEvent
    /rosout: rcl_interfaces/msg/Log
    /turtle1/color_sensor: turtlesim/msg/Color
    /turtle1/pose: turtlesim/msg/Pose
    /turtle1/rotate_absolute/_action/feedback: turtlesim/action/RotateAbsolute_FeedbackMessage
    /turtle1/rotate_absolute/_action/status: action_msgs/msg/GoalStatusArray
  Services:
    /clear: std_srvs/srv/Empty
    /kill: turtlesim/srv/Kill
    /reset: std_srvs/srv/Empty
    /spawn: turtlesim/srv/Spawn
    /turtle1/rotate_absolute/_action/cancel_goal: action_msgs/srv/CancelGoal
    /turtle1/rotate_absolute/_action/get_result: turtlesim/action/RotateAbsolute_GetResult
    /turtle1/rotate_absolute/_action/send_goal: turtlesim/action/RotateAbsolute_SendGoal
    /turtle1/set_pen: turtlesim/srv/SetPen
    /turtle1/teleport_absolute: turtlesim/srv/TeleportAbsolute
    /turtle1/teleport_relative: turtlesim/srv/TeleportRelative
    /turtlesim/describe_parameters: rcl_interfaces/srv/DescribeParameters
    /turtlesim/get_parameter_types: rcl_interfaces/srv/GetParameterTypes
    /turtlesim/get_parameters: rcl_interfaces/srv/GetParameters
    /turtlesim/list_parameters: rcl_interfaces/srv/ListParameters
    /turtlesim/set_parameters: rcl_interfaces/srv/SetParameters
    /turtlesim/set_parameters_atomically: rcl_interfaces/srv/SetParametersAtomically
  Action Servers:
    /turtle1/rotate_absolute: turtlesim/action/RotateAbsolute
  Action Clients:
```

Now, we can get the same output if we add the `--include-hidden` option, but by default we don't see the hidden entities like `/turtle1/rotate_absoulte/_action/cancel_goal`:

```
/turtlesim
  Subscribers:
    /parameter_events: rcl_interfaces/msg/ParameterEvent
    /turtle1/cmd_vel: geometry_msgs/msg/Twist
  Publishers:
    /parameter_events: rcl_interfaces/msg/ParameterEvent
    /rosout: rcl_interfaces/msg/Log
    /turtle1/color_sensor: turtlesim/msg/Color
    /turtle1/pose: turtlesim/msg/Pose
  Services:
    /clear: std_srvs/srv/Empty
    /kill: turtlesim/srv/Kill
    /reset: std_srvs/srv/Empty
    /spawn: turtlesim/srv/Spawn
    /turtle1/set_pen: turtlesim/srv/SetPen
    /turtle1/teleport_absolute: turtlesim/srv/TeleportAbsolute
    /turtle1/teleport_relative: turtlesim/srv/TeleportRelative
    /turtlesim/describe_parameters: rcl_interfaces/srv/DescribeParameters
    /turtlesim/get_parameter_types: rcl_interfaces/srv/GetParameterTypes
    /turtlesim/get_parameters: rcl_interfaces/srv/GetParameters
    /turtlesim/list_parameters: rcl_interfaces/srv/ListParameters
    /turtlesim/set_parameters: rcl_interfaces/srv/SetParameters
    /turtlesim/set_parameters_atomically: rcl_interfaces/srv/SetParametersAtomically
  Action Servers:
    /turtle1/rotate_absolute: turtlesim/action/RotateAbsolute
  Action Clients:
```

### CODE REVIEW

File: ros2node/ros2node/api/__init__.py [hunk 1]
```diff
@@ -57,48 +65,51 @@ def get_node_names(*, node, include_hidden_nodes=False):
...
             types=t[1])
-        for t in names_and_types]
+        for t in names_and_types if include_hidden_topics or not _is_hidden_name(t[0])]
 
 
-def get_subscriber_info(*, node, remote_node_name):
-    return get_topics(remote_node_name, node.get_subscriber_names_and_types_by_node)
+def get_subscriber_info(*, node, remote_node_name, include_hidden=False):
+    return get_topics(
+        remote_node_name, node.get_subscriber_names_and_types_by_node, include_hidden)
```
Comment (dirk-thomas) [inline ros2node/ros2node/api/__init__.py]: Keyword arguments should be passed with the explicit keyword. Same below.

Maybe the called function should force that with `, *, include_hidden_topics=False`.

File: ros2node/ros2node/api/__init__.py [hunk 2]
```diff
@@ -24,6 +24,14 @@
 TopicInfo = namedtuple('Topic', ('name', 'types'))
 
 
+def _is_hidden_name(name):
+    # note, we're assuming the hidden node prefix is the same for other hidden names
+    return (
+        name and
```
Comment (dirk-thomas) [inline ros2node/ros2node/api/__init__.py]: What is this check needed for?

Comment (jacobperron) [reply ros2node/ros2node/api/__init__.py]: It was copied from the [hidden node logic](https://github.com/ros2/ros2cli/blob/c17f7dbb7fb8d15046b8719e22a2fb71e6c3e9f8/ros2node/ros2node/api/__init__.py#L54), but I guess we can't have empty topic names so this shouldn't be needed.

Comment (jacobperron) [reply ros2node/ros2node/api/__init__.py]: Removed 2367227

File: ros2node/ros2node/api/__init__.py (hunk 1 above)
Comment (jacobperron) [reply ros2node/ros2node/api/__init__.py]: Done: 2367227

**Comment by jacobperron** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8506/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4390/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6922/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8415/)

File: ros2node/ros2node/verb/info.py [hunk 3]
```diff
@@ -38,28 +38,34 @@ def add_arguments(self, parser, cli_name):
             'node_name',
             help='Node name to request information')
         argument.completer = NodeNameCompleter()
+        parser.add_argument(
+            '--include-hidden', action='store_true',
+            help='Display hidden topics, services, and actions as well')
```
Comment (dirk-thomas) [inline ros2node/ros2node/verb/info.py]: Maybe a user wants to see hidden nodes but not hidden topics / services / actions - or the other way around? Just a thought...

Comment (jacobperron) [reply ros2node/ros2node/verb/info.py]: I presume you mean them wanting to see hidden topics, but not hidden services / actions (or some combination). I thought about it, but I figured it's a rare case to want to see any of the hidden names, let alone a subset.

**Pull request merged**